[6454] dont clear chat history on deletion

bhauman opened the pull request:
fixes #6454  

### Summary:

Summary:

If a public chat is removed from the chat list and then reopened again, its content is empty. There is "Fetching messages" but the old messages (before removing the chat) do not appear. I think it's counter-intuitive and the app should display old messages when public chat is re-opened.
I'd say old messages should be displayed even though the user has cleared the chat before removing it. So basically joining/re-opening a public chat should fetch all messages.

Reproduction
Join public chat with some messages
Remove it
Join the same public chat again
User should see historical messages (currently 7 days of history) which is not the case

### Review notes (optional):
This is my first PR on a issue on status-react. So I'm not familiar with all the ramifications here but this appears to solve it. 

It seems like `clear-history` is miss-used here.

status: ready

Commit:
[6454] dont clear chat history on deletion

ghost commented:
Hey @bhauman, and thank you so much for making your first pull request in status-react! :heart: Please help us make your experience better by filling out this brief questionnaire https://goo.gl/forms/uWqNcVpVz7OIopXg2

<blockquote><img src="https://lh5.googleusercontent.com/wpBxRKA0YohehKsTaptQsaYJNVcNA9m2IOLvxXpSMcamdM8e4q7gyYBQ_yRVvNH-Ez8=w1200-h630-p" width="48" align="right"><div><img src="//ssl.gstatic.com/docs/forms/device_home/android_192.png" height="14"> Google Docs</div><div><strong><a href="https://docs.google.com/forms/d/e/1FAIpQLSewRMgMD6FLe9yO7oCipVn7YTn1HboJc4ZDJ_Va_BsDAemQwQ/viewform?usp=send_form&usp=embed_facebook">Status 1st commit dev survey</a></strong></div><div>First of all thank you so much for taking the time to make your first foray into the world of Status. We want to make sure that your experience with building the future of the decentralized web with us is not only enjoyable but gives you real purpose.

 Please check out our wiki at https://wiki.status.im and our Dev forum at https://discuss.status.im for more in depth longform discussion -- You can also join our Riot at https://chat.status.im which has all of our developers to answer any pressing questions you may have.</div></blockquote>

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status.html))?

bhauman commented:
The real problem is that clear-history changes the `:deleted-at-clock-value` and that is not what is called for in this case.  However it does not seem necessary to actually delete the messages at all so, just skip it. 

If clean up is actually needed then it seems that we just need to call `messages-store/delete-messages-tx` on the chat id not move the `:deleted-at-clock-value`

chadyj commented:
Congrats on your first PR!  

I left a comment over in the issue https://github.com/status-im/status-react/issues/6454#issuecomment-432121014 that might be worth discussing.

bhauman commented:
According to the discussion over on https://github.com/status-im/status-react/issues/6454#issuecomment-432121014

Not clearing the history is not the solution here I do think that simply deleting the messages and not moving the `deleted-at-clock-value` meets both the security needs and UX needs.

bhauman commented:
So I'm going to close this PR and go back to the drawing board. And I'll fix the tests on that one ;)
